icu4c/source/i18n/chnsecal.h

```diff
@@ -152,7 +195,12 @@ class U_I18N_API ChineseCalendar : public Calendar {
   // Internal data....
   //-------------------------------------------------------------------------
     
-  UBool isLeapYear;
+  // There is a leap month between the Winter Solstice before and after the
+  // current date.This is different from leap year because in some year, such as
+  // 1813 and 2033, the leap month is after the Winter Solstice of that year. So
+  // this value could be false for a date prior to the Winter Solstice of that
+  // year but that year still has a leap month and therefor is a leap year.
+  UBool hasLeapMonthBetweenWinterSolstices;
```

Comment:
so the issue is in Chinese calendar, the calculation of leap is actually not determined by a 'year' (Chinese Jan-1 to Dec-30) but determined by a Sui (between two Winter Solstices) . The variable used to call isLeapYear which is misleading because that boolean value is used to indicate there is a leap month in the current "sui" (but the begin/end of a "sui" is not the begin/end of a "year") but not necessary in the current "year".

Notice "sui" is not a Chinese "year". So we have three (not two) different concepts
1) a Gregorian Year
2) a Chinese Year (12 or 13 months)- which go with the Moon.
3) a Sui (the period between two Winter Solstices,  Winter Solstices are usually on Gregorian Dec 21 or Dec 22, and go with the Sun, not Moon)

That variable is indicating the particular moment (the date value) is in a Sui which has a leap month. We could name that variable inLeapSui but that is not the same as inLeapYear.

The ECMA Temporal API asked us to return the answer of it is in a leap YEAR or not, not in a leap Sui or not, so we cannot just return that value. 

There are dates in a Leap year which is NOT in a leap Sui. For example, If we have a leap April in Chinese year, any date in the next year is always not in a leap year (since there are not consective leap year in Chinese calendar), and any dates after the Winter Solstices (Gregorian Calendar Dec 21 or 22, always a date in Chinese calendar the "11th month" since Chinese calendar guarantee Winter Solstices fall into Chinese "11th month" ) of that year will be still in the Leap Year, but not in the leap Sui. Similarly, For the date in the previous year after the Winter Solstices, that date (in the year prior to the one have Leap April) will be in non leap year but that variable will be true because there is a leap month between that two Winter Solstices.

Reply:
Thank you for that explanation; that was very clear and answers my questions.

To make sure I understand, what you're saying is that a calendar year in the Chinese year straddles two "sui"s, either of which (but not both) may contain a leap month.  If either does, the calendar year can be considered a "leap year."  (Or does the leap month itself have to be within that calendar year?  And is that the Chinese-calendar year the "is leap year" stuff is talking about, or the corresponding Gregorian year?)  The `hasLeapMonthBetweenWinterSolstices()` method tells you whether the "sui" containing the current date has a leap month, but if it returns false, the calendar year might still be a "leap year" because the _other_ "sui" might contain a leap month.

Do I have this right?

Your explanation is great, but long.  I wonder if it's possible to condense it a little and put that into the source code somewhere?  Alternatively, there might be a better name for the function, but I can't think of one.

Reply:
"If either does, the calendar year can be considered a "leap year.""

NO. the leap year cannot be derived from the leap sui. Look at the following
```
-----WS---G----C--------------------------------WS---G----C-------------------
-----|-------------------Sui--------------------|-----------------------------
-Non Leap Year-|--------------Leap Year-------------------|-Non Leap Year-----
1      2                 3                           4           5
```
where 
* WS is the Winter Solstice
* G is the start of a Gregoria Year
* C is the start of a Chinese year 

A date is in a leap year if there is a leap month between two Cs, 
That variable indicate there is a leap month between two WSs.
so the condiction is not just "If either does". It is acutally checking two different thing. 
The leap month may happen in point 3 (from the beginning of the year till WS) or point 4 (between WS to the end of the year). It is very rare for it to happen in point 4 but still possible.
That varialbe only indicate there is a leap month fall into the current Sui. 
If the date value is in 1 or 2, (but the leap month is in point 3 or 4), it is NOT in a leap year
For example, if a leap month fall into point 3, then point 2 and point 3 are both in the same Sui, but point 2 is not in a leap year but point 3 is in a leap year. and the value of that variable is true for both point 2 and point 3. 

Also, for some rarely happen year (for example 2033), the leap month may happen in point 4, in that case, both point 3 and point 4 are in leap year but not point 1 , 2 nor 5. but for that variable (hasLeapMonthBetweenWinterSolstices) , it is true in point 4 and 5 but false in point 1, 2, and 3.

Reply:
Okay, that all makes sense.  Thanks again.